test/fixtures/dialects/ansi/insert_with_statement.sql

```diff
@@ -0,0 +1,13 @@
+INSERT INTO table2 (column1, column2, column3)
```

Comment:
Shouldn't the INSERT be after the CTE?

Reply:
So this was already in the logic prior to this change, Bbut interestingly this is correct behaviour. Was new to me too 😄 

Postgres (from https://www.postgresql.org/docs/9.5/sql-insert.html):
![image](https://user-images.githubusercontent.com/80432516/141660189-61075a3f-99d6-4591-b496-6504a154c156.png)

Similar for Snowflake (https://community.snowflake.com/s/question/0D50Z00008EM1Qu/does-snowflake-support-ctewith-clause-on-an-insert-statement):
![image](https://user-images.githubusercontent.com/80432516/141660284-ad5b95c0-b4a2-4e8d-9fa0-52b0a50cf340.png)

MySQL too! (https://stackoverflow.com/questions/24008316/insert-into-from-cte):
![image](https://user-images.githubusercontent.com/80432516/141660327-bab91edc-9991-4519-90d4-aab95828da50.png)

So I assume the decision was made in the past to include with the ANSI dialect 👍

Reply:
Fair enough - makes sense!